src/prog_algs/state_estimators/particle_filter.py

```diff
@@ -80,7 +81,7 @@ def __measure(x):
     def __str__(self):
         return "{} State Estimator".format(self.__class__)
         
-    def estimate(self, t, u, z):
+    def estimate(self, t : float, u : dict, z : dict):
```

Comment:
```suggestion
    def estimate(self, t : float, u :"InputContainer", z :"OutputContainer"):
```
Would this work?

Reply:
Can you also update the doc comments here and for kalman filter

Reply:
I don't think this works - Python evaluates the string to a type and because InputContainer/OutputContainer are not actual types, this fails